packages/devtools_app/lib/src/debugger/flutter/debugger_screen.dart

```diff
@@ -60,13 +73,128 @@ class DebuggerScreenBodyState extends State<DebuggerScreenBody> {
       axis: Axis.horizontal,
       initialFirstFraction: 0.25,
       // TODO(https://github.com/flutter/devtools/issues/1648): Debug panes.
-      firstChild: const Text('Debugger details'),
+      firstChild: ScriptPicker(
+        scripts: scriptList,
+        onSelected: onScriptSelected,
+        selected: loadingScript,
+      ),
       // TODO(https://github.com/flutter/devtools/issues/1648): Debug controls.
-      secondChild: CodeView(
-        script: script,
+      secondChild: loadingScript != null && script == null
+          ? const Center(child: CircularProgressIndicator())
+          : CodeView(
+              script: script,
+            ),
+    );
+  }
+}
+
+/// Picker that takes a [ScriptList] and allows selection of one of the scripts inside.
+class ScriptPicker extends StatefulWidget {
+  const ScriptPicker({Key key, this.scripts, this.onSelected, this.selected})
+      : super(key: key);
+
+  final ScriptList scripts;
+  final void Function(ScriptRef scriptRef) onSelected;
+  final ScriptRef selected;
+
+  @override
+  ScriptPickerState createState() => ScriptPickerState();
+}
+
+class ScriptPickerState extends State<ScriptPicker> {
+  List<ScriptRef> _filtered;
+  TextEditingController filterController = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    if (_isNotLoaded) initFilter();
+  }
+
+  @override
+  void didUpdateWidget(ScriptPicker oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (_isNotLoaded) {
+      initFilter();
+    } else if (oldWidget.scripts != widget.scripts) {
+      updateFilter(filterController.text);
+    }
+  }
+
+  void initFilter() {
+    // Make an educated guess as to the main package to slim down the initial list of scripts we show.
+    if (widget.scripts?.scripts != null) {
+      final mainFile = widget.scripts.scripts
+          .firstWhere((ref) => ref.uri.contains('main.dart'));
+      filterController.text = mainFile.uri.split('/').first;
+      updateFilter(filterController.text);
+    }
+  }
+
+  void updateFilter(String value) {
+    setState(() {
+      if (widget.scripts?.scripts == null) {
+        _filtered = null;
+      } else {
+        _filtered = widget.scripts.scripts
+            .where(
+              (ref) => ref.uri.contains(value.toLowerCase()),
+            )
+            .toList();
+      }
+    });
+  }
+
+  Widget _buildScript(ScriptRef ref) {
+    final selectedColor = Theme.of(context).selectedRowColor;
+    return Material(
+      color: ref == widget.selected ? selectedColor : null,
+      child: InkWell(
+        onTap: () => widget.onSelected(ref),
+        child: Padding(
+          padding: const EdgeInsets.all(4.0),
+          child: Align(
+            alignment: Alignment.centerLeft,
+            child: Text('${ref?.uri?.split('/')?.last} (${ref?.uri})'),
```

Comment:
this seems a bit different than what html_scripts_view.dart is currently showing. Can you make it consistent or add a TODO to make it consistent with a followup cl?
Specifically: matching text should be in bold and we should be showing the entire uri.
What we have currently in the html app isn't perfect so it could also be reasonable to iterate with UXR to figure out what we really want to show here.

Reply:
I came to this design from meeting with internal users last week.  They explained that one of the problems with the existing debugger page is that they can't see the names of files in the list because there is too much boilerplate in the long g3 package paths.  To remedy this, I've adopted the form:

```
foo.dart (package:my_long_package_name/lib/src/long/path/to/foo.dart)
```

> DBC: can you put a border and padding around the two sections to break up the view? It is a bit difficult to visually parse the UI and the two sections are too close to the splitter grippy.
> 
> Example of what we do for the flame chart:
> 
> ```
>     Padding(
>       padding: const EdgeInsets.only(bottom: 8.0),
>       child: Container(
>         key: TimelineScreen.flameChartSectionKey,
>         decoration: BoxDecoration(
>           border: Border.all(color: Theme.of(context).focusColor),
>         ),
>         child: content,
>       ),
>     );
> ```

Good idea; added this as a common widget.

Reply:
I had also hoped users would want to use package: for this but that isn't what our users say overall.
we did a user survey on this last summer and 38% of users want to see package: and 39% of users prefer package: while 56% percent prefer file paths relative to their app route.
in bazel projects the package: urls are particularly bad and counterintuitive. Rather than
package:my_long_package_name/lib/src/long/path/to/foo.dart
you get
package:my_long_package_name.src.long.path.to/foo.dart

Reply:
maybe just add a TODO resolve this. For example, you could wrap each file its own scrollable so the filename was always visible and the start of the path was truncated. Would be good to get UXR involved on this as there are some tradeoffs and there is a lot of plcaes in the tool where we would like to do a better job of displaying file names.